conda_vendor/conda_vendor.py

```diff
@@ -1,218 +1,499 @@
+                repo_data["packages"][name] = entry
+            pb.update(1)
 
-# see https://stackoverflow.com/questions/21371809/cleanly-setting-max-retries-on-python-requests-get-or-post-method
-def improved_download(url):
-    session = requests.Session()
-    retry = Retry(connect=5, backoff_factor=0.5)
-    adapter = HTTPAdapter(max_retries=retry)
-    session.mount("http://", adapter)
-    session.mount("https://", adapter)
-    return session.get(url)
+        for name, entry in _packages_dot_conda.items():
+            if name in valid_names:
+                repo_data["packages.conda"][name] = entry
+            pb.update(1)
 
-def download_solved_pkgs(fetch_action_pkgs, vendored_path, platform):
-    click.echo(click.style("Downloading and Verifying SHA256 Checksums for Solved Packages", bold=True, fg='green'))
+    # write to destination
+    dest_file = Path(f"{dest_dir}/repodata.json")
+    with dest_file.open("w") as f:
+        json.dump(repo_data, f)
 
-    def _download_solved_pkgs(pkg, vendored_path, platform):
-        platform_path = vendored_path / platform
-        response = improved_download(pkg['url'])
-        content = response.content
 
-        # verify checksum
-        compare_sha256(content, pkg['sha256'])
-        with open(platform_path / pkg['fn'], "wb") as conda_pkg:
-            conda_pkg.write(content)
+# hotfix vendored repodata.json given the input of FETCH action packages
+# from conda-lock's solve results
+def hotfix_vendored_repodata_json(
+    package_list: List[FetchAction], vendored_dir_path: Path
+):
+    """Go through the package_list, i.e. the solution provided by conda_lock,
+    and generate a new repodata.json at vendored_dir_path/{subdir}, where
+    subdir is either noarch or the platform (as specified in the metadata
+    in package_list).
+
+    Parameters
+    ----------
+    package_list: list [ conda_lock.conda_solver.FetchAction ]
+        list of packages (and their metadata) provided by conda-lock
+
+    vendored_dir_path: pathlib.Path
+        location of the root of the new conda channel
+    """
+    channels = []
+    subdirs = []
 
-    with click.progressbar(fetch_action_pkgs, label="Downloading Progress") as pkgs:
+    for pkg in package_list:
+        channels.append(pkg["channel"])
+        subdirs.append(pkg["subdir"])
+
+        _blue(70 * "=")
+        _yellow(f"Channel: {pkg['channel']}", bold=False)
+        _yellow(f"Package: {pkg['fn']}", bold=False)
+        _yellow(f"URL: {pkg['url']}", bold=False)
+        _yellow(f"SHA256: {pkg['sha256']}", bold=False)
+        _yellow(f"Subdirectory: {pkg['subdir']}", bold=False)
+        _yellow(f"Timestamp: {pkg['timestamp']}", bold=False)
+        _blue(70 * "=")
+
+    channels = list(set(channels))
+    subdirs = list(set(subdirs))
+
+    # patch repodata.json for each channel + subdir
+
+    # this monstrosity just call reconstruct_repodata for each subdir like
+    # "noarch" , "osx-64", "linux-64", with each channel like
```

Comment:
just "calls"? instead of "call"

Reply:
it's in a comment, i'll fix it because i'm fixing "etry", but otherwise, i don't care.

Reply:
fixed